client.go

```diff
@@ -1,23 +1,39 @@
 package typhon
 
 import (
+	"context"
+	"crypto/tls"
+	"net"
 	"net/http"
 	"time"
 
 	"github.com/monzo/terrors"
+	"golang.org/x/net/http2"
 )
 
 var (
 	// Client is used to send all requests by default. It can be overridden globally but MUST only be done before use
 	// takes place; access is not synchronised.
 	Client Service = BareClient
 	// RoundTripper is used by default in Typhon clients
-	RoundTripper http.RoundTripper = &http.Transport{
+	// It chooses HTTP1, H2C or HTTP2 based on a context flag (see WithHTTP2)
+	RoundTripper http.RoundTripper = dynamicRoundTripper{}
+	// HTTP1RoundTripper is a HTTP1 client
+	HTTP1RoundTripper http.RoundTripper = &http.Transport{
```

Comment:
Is this definitely a HTTP/1.1 only round tripper? The default `http.Transport` will upgrade to HTTP/2 unless custom TLS config or custom dialers have been set I think?

https://golang.org/src/net/http/transport.go?s=3397:10477#L353
```go
if !t.ForceAttemptHTTP2 && (t.TLSClientConfig != nil || t.Dial != nil || t.DialTLS != nil || t.DialContext != nil) {
```
So in this case `ForceAttemptHTTP2` will default to `false`, but we've not overridden any of the others, so it might still upgrade?

Reply:
Yeah, you're right actually, HTTP2 upgrade is already supported. I think we only need an H2C client and can change to `WithH2C`